source/Octopus.Tentacle/Kubernetes/KubernetesModule.cs

```diff
@@ -28,6 +31,16 @@ protected override void Load(ContainerBuilder builder)
             builder.RegisterType<KubernetesDirectoryInformationProvider>().As<IKubernetesDirectoryInformationProvider>().SingleInstance();
             builder.RegisterType<MemoryCache>().As<IMemoryCache>().SingleInstance();
             builder.RegisterType<MemoryCacheOptions>().As<IOptions<MemoryCacheOptions>>().SingleInstance();
+            builder.RegisterType<PersistenceProvider>()
+                .Named<IPersistenceProvider>("KubernetesAgentMetricsConfigMap")
+                .WithParameter("configMapName", "kubernetes-agent-metrics");
+            builder.RegisterType<KubernetesAgentMetrics>().AsSelf().SingleInstance()
+                .WithParameter(
+                    new ResolvedParameter(
```

Comment:
Is it necessary to use a named registration in this case given that there is only one implementation?
Also what happens if the name of the PersistenceProvider doesn't line up, will an exception be thrown at runtime when it fails to resolve?

Reply:
Yeah - named is a bit much (YAGNI).

It started with not wanting hard-coded names in the ConfigMap - as that impedes integration testing.
Creating a named item _seemed_ to make things nicer as you can now say "KubernetesAgentMetrics depends on config-map-X" - while capability-Y depends on configh-map-y.

Reply:
updated following the informal code review to use delegate creation factories in autofac.